pkg/reconciler/image/build_required.go

```diff
@@ -0,0 +1,137 @@
+package image
+
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/pivotal/kpack/pkg/apis/build/v1alpha1"
+	"github.com/pivotal/kpack/pkg/buildchange"
+)
+
+type buildRequiredResult struct {
+	ConditionStatus corev1.ConditionStatus
+	ReasonsStr      string
+	ChangesStr      string
+}
+
+func newBuildRequiredResult(summary buildchange.ChangeSummary) buildRequiredResult {
+	var result buildRequiredResult
+	if summary.HasChanges {
+		result.ConditionStatus = corev1.ConditionTrue
+	} else {
+		result.ConditionStatus = corev1.ConditionFalse
+	}
+	result.ReasonsStr = summary.ReasonsStr
+	result.ChangesStr = summary.ChangesStr
+	return result
+}
+
+func isBuildRequired(img *v1alpha1.Image,
+	lastBuild *v1alpha1.Build,
+	srcResolver *v1alpha1.SourceResolver,
+	builder v1alpha1.BuilderResource) (buildRequiredResult, error) {
+
+	result := buildRequiredResult{ConditionStatus: corev1.ConditionUnknown}
+	if !srcResolver.Ready() || !builder.Ready() {
+		return result, nil
+	}
+
+	changeSummary, err := buildchange.NewChangeProcessor().
+		Process(triggerChange(lastBuild)).
+		Process(commitChange(lastBuild, srcResolver)).
+		Process(configChange(img, lastBuild, srcResolver)).
+		Process(buildpackChange(lastBuild, builder)).
+		Process(stackChange(lastBuild, builder)).
+		Summarize()
+	if err != nil {
+		return result, err
+	}
+
+	return newBuildRequiredResult(changeSummary), nil
+}
+
+func triggerChange(lastBuild *v1alpha1.Build) buildchange.Change {
+	if lastBuild == nil || lastBuild.Annotations == nil {
+		return nil
+	}
+
+	time, ok := lastBuild.Annotations[v1alpha1.BuildNeededAnnotation]
+	if !ok {
+		return nil
+	}
+	return buildchange.NewTriggerChange(time)
+}
+
+func commitChange(lastBuild *v1alpha1.Build, srcResolver *v1alpha1.SourceResolver) buildchange.Change {
+	// If the lastBuild was not a Git source, then it is not a COMMIT change
+	if lastBuild == nil || lastBuild.Spec.Source.Git == nil || srcResolver.Status.Source.Git == nil {
```

Comment:
Is the git commit still displayed when the lastBuild was not a git source build but, the current build is a git commit source?

Reply:
>  when the lastBuild was not a git source build but, the current build is a git commit source

the above is treated as a CONFIG change and not a COMMIT change

changes are marked as COMMIT happens when the only diff is the git revision.

Reply:
```
$kp image create test-image --tag gcr.io/cf-build-service-dev-219913/ssuresh/test-image --local-path /Users/ssuresh/workspace/samples/go/mod
Creating Image...
Uploading to 'gcr.io/cf-build-service-dev-219913/ssuresh/test-image-source'...
	Uploading 'gcr.io/cf-build-service-dev-219913/ssuresh/test-image-source@sha256:99bbe5fac447867858669d17471295387056e28fed2fe4bd9e8403ae8bd51759'
Image "test-image" created

$ kp build logs test-image
===> PREPARE
Build reason(s): CONFIG
CONFIG:
	resources: {}
	- source: {}
	+ source:
	+   registry:
	+     image: gcr.io/cf-build-service-dev-219913/ssuresh/test-image-source@sha256:99bbe5fac447867858669d17471295387056e28fed2fe4bd9e8403ae8bd51759
Loading secret for "gcr.io" from secret "gcr" at location "/var/build-secrets/gcr"
...

$ kp image patch test-image --git  https://github.com/paketo-buildpacks/samples --git-revision main --sub-path go/mod
Patching Image...
Image "test-image" patched

$ kp build logs test-image
===> PREPARE
Build reason(s): CONFIG
CONFIG:
	resources: {}
	source:
	-   registry:
	-     image: gcr.io/cf-build-service-dev-219913/ssuresh/test-image-source@sha256:99bbe5fac447867858669d17471295387056e28fed2fe4bd9e8403ae8bd51759
	+   git:
	+     revision: d6046f84d39aee034a69cc4dc1e5d33b06a651eb
	+     url: https://github.com/paketo-buildpacks/samples
	+   subPath: go/mod
Loading secret for "gcr.io" from secret "gcr" at location "/var/build-secrets/gcr"
...
```